devito/types/sparse.py

```diff
@@ -742,7 +745,11 @@ def __shape_setup__(cls, **kwargs):
                 raise ValueError('`nt` must be > 0')
 
             shape = list(AbstractSparseFunction.__shape_setup__(**kwargs))
-            shape.insert(cls._time_position, nt)
+            try:
+                # Rebuilt with full shape
+                assert shape[cls._time_position] == nt
+            except AssertionError:
+                shape.insert(cls._time_position, nt)
```

Comment:
Im also not sure I understand when can we end up here? perhaps it's simply because I don't understand this bug yet... and this is entirely on me

Reply:
We end up here when we create a SparseTimeFunction which uses `nt` to infer the time size. At rebuild it has both nt and shape. Can just ignore shape for time function but that'd still need a try/except